File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum = 0;
    void dfs(TreeNode root){
    //base case
    if(root == null){
        return;
    }
    // recursive case
    dfs(root.right);
    sum += root.val;
    root.val = sum;
    dfs(root.left);
}
    public TreeNode bstToGst(TreeNode root) {
        // Code with Radheshyam (.^.)
        dfs(root);
        return root;
    }
}

File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
int sum=0;
void dfs(TreeNode *root){
    //base case
    if(root == NULL){
        return;
    }
    // recursive case
    dfs(root->right);
    sum += root->val;
    root->val = sum;
    dfs(root->left);
}
    TreeNode* bstToGst(TreeNode* root) {
        // Code with Radheshyam (.^.)
        dfs(root);
        return root;
    }
};